### ISSUE TITLE
History does not honor Group Security when someone is added to a group

### ISSUE THREAD

**Issue by atamburino** [SparkDevNetwork/Rock#5332]
### Please go through all the tasks below ###

* [x] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side. ###

Similar to https://github.com/SparkDevNetwork/Rock/issues/5043 & https://github.com/SparkDevNetwork/Rock/issues/5043.

When viewing the attendance history for a person, if they have been added to that the current user is not authorized to view, they can still see that the user was added to that group. This is a sensitive issue for people who may be attending a recovery-type group or other care-focused groups.

![image](https://user-images.githubusercontent.com/57277787/222755074-a4fd498a-a047-45e2-a56f-56ab1a969e92.png)

### Expected Behavior ###

Because the member we impersonate is not in the Rock Administration role, they should not see that the new member was added to the care/recovery group.

### Actual Behavior ###

The none admin can see that the new member was added to a recovery/care group.

### Steps to Reproduce ###

1. Go to a site running Rock McKinley 14.1 (1.14.1.1).
1. Create a new member and add them to a group that has a security set where only an admin security role can view."
1. Impersonate any other member that can see the history section on a person's profile page but not the group
1. As this person, view the history tab on the new member you created on step 3's profile

### Rock Version ###

Rock McKinley 14.1 (1.14.1.1)

### Client Culture Setting ###

Client Culture Setting: en-US

**Comment by bobrufenacht** [comment]
Note that this issue is not just a duplicate of #5043.  The issue remains for Group Security but it is also an issue with Attendance as attendance in those groups may be displayed as well.

**Comment by tpeera** [comment]
@Kwame-Agyei would a practical work around for earlier versions prior to v16.1 be to use the **Person History** block under the category CRM > Person Detail? _(and then secure the History Log block until v16.1)_

**Comment by Kwame-Agyei** [comment]
Hello @[tpeera](https://github.com/tpeera),
 
Well, the PersonHistory block is now considered outdated and hasn't been updated for quite some time. It basically serves the same function as the HistoryLog block on the Person history page, but there's a chance it might have bugs. If you plan to switch to it, we would advise you to make sure to set a reminder to remove or replace the block once you upgrade to v16.1.
